urlc4364d0d12_ahmedelmorsy_cafetria_simulation_tgz-pJ8-main_MainJ8/src/simulation/global/Statistics.java

```diff
@@ -0,0 +1,370 @@
+        temp.add(tempInfo);
+        queuesInfoMap.put(Const.CASHIER_SERVER, temp);
+
+        delaysInfoMap = new HashMap<String, ArrayList<Delay>>();
+        ArrayList<Delay> list = new ArrayList<Statistics.Delay>();
+        delaysInfoMap.put(Const.HOTFOOD_SERVER, list);
+        list = new ArrayList<Statistics.Delay>();
+        delaysInfoMap.put(Const.SANDWICH_SERVER, list);
+        list = new ArrayList<Statistics.Delay>();
+        delaysInfoMap.put(Const.DRINKS_SERVER, list);
+        list = new ArrayList<Statistics.Delay>();
+        delaysInfoMap.put(Const.CASHIER_SERVER, list);
+
+        tempDelays = new HashMap<Customer, Statistics.Delay>();
+
+        timeAvgNumInQueue = new HashMap<String, Double>();
+        maxNumInQueue = new HashMap<String, Integer>();
+        avgDelayInQueue = new HashMap<String, Double>();
+        maxDelayInQueue = new HashMap<String, Integer>();
+        custDelay = new HashMap<Customer, Integer>();
+    }
+
+    public static void drawQueueLen() {
+        final JFreeChart chart = ChartFactory.createXYLineChart(
+                "Line Chart Demo 6", // chart title
+                "X", // x axis label
+                "Y", // y axis label
+                new XYSeriesCollection(queueLengthDataset), // data
+                PlotOrientation.VERTICAL, true, // include legend
+                true, // tooltips
+                false // urls
+                );
+        chart.setBackgroundPaint(Color.white);
+
+        // final StandardLegend legend = (StandardLegend) chart.getLegend();
+        // legend.setDisplaySeriesShapes(true);
+
+        // get a reference to the plot for further customisation...
+        final XYPlot plot = chart.getXYPlot();
+        plot.setBackgroundPaint(Color.lightGray);
+        // plot.setAxisOffset(new Spacer(Spacer.ABSOLUTE, 5.0, 5.0, 5.0, 5.0));
+        plot.setDomainGridlinePaint(Color.white);
+        plot.setRangeGridlinePaint(Color.white);
+
+        final XYLineAndShapeRenderer renderer = new XYLineAndShapeRenderer();
+        renderer.setSeriesLinesVisible(0, false);
+        renderer.setSeriesShapesVisible(1, false);
+        plot.setRenderer(renderer);
+
+        // change the auto tick unit selection to integer units only...
+        final NumberAxis rangeAxis = (NumberAxis) plot.getRangeAxis();
+        rangeAxis.setStandardTickUnits(NumberAxis.createIntegerTickUnits());
+
+        final ChartPanel chartPanel = new ChartPanel(chart);
+        chartPanel.setPreferredSize(new java.awt.Dimension(500, 270));
+        ApplicationFrame demo = new ApplicationFrame("chart");
+        demo.setContentPane(chartPanel);
+        demo.pack();
+        RefineryUtilities.centerFrameOnScreen(demo);
+        demo.setVisible(true);
+
+    }
+
+    public static void UpdateQueueLength(int len, String queueType) {
+        ArrayList<QueueInfo> temp = queuesInfoMap.get(queueType);
+        QueueInfo info = temp.get(temp.size() - 1);
+        if (queueType == Const.HOTFOOD_SERVER) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.